llvm/lib/SYCLPostLink/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+add_llvm_component_library(LLVMSYCLPostLink
```

Comment:
about the name, to me syclpostlink is a tool, so i would expect files to be in `clang/tools`, so maybe we could rename it to something else, but i don't have any great ideas, `SYCLLowering`, `SYCLFinalization`, idk

Reply:
`sycl-post-link`'s help text reads:
> This is a collection of utilities run on device code's LLVM IR before handing off to back-end for further compilation or emitting SPIRV.

... so I think the verb "finalize" would capture that well. `SYCLFinalizeIR` in analogy to `SYCLLowerIR` maybe?

I also wouldn't mind keeping the "post-link" moniker. `SYCLPostLinkUtils` or so would make it clear that this is not a tool.

Reply:
`sycl-post-link` tool is going to disappear after migration to New Offload Model.
`clang/tools` is not appropriate since `sycl-jit` and intel's fortran compiler depend on this functionality.